codegen/templates/operator/main.tmpl

```diff
@@ -37,59 +35,51 @@ func main() {
 		panic(err)
     }
 
-    runner, err := simple.NewOperator(simple.OperatorConfig{
-		Name:       "{{.ProjectName}}-operator",
-		KubeConfig: kubeConfig.RestConfig,
-		Metrics: simple.MetricsConfig{
-			Enabled: true,
-		},
-		Tracing: simple.TracingConfig{
-			Enabled: true,
-			OpenTelemetryConfig: simple.OpenTelemetryConfig{
-				Host:        cfg.OTelConfig.Host,
-				Port:        cfg.OTelConfig.Port,
-				ConnType:    simple.OTelConnType(cfg.OTelConfig.ConnType),
-				ServiceName: cfg.OTelConfig.ServiceName,
-			},
-		},
-		ErrorHandler: func(ctx context.Context, err error) {
-            // FIXME: add your own error handling here
-			logging.FromContext(ctx).Error(err.Error())
-		},
-	})
-    if err != nil {
-        logging.DefaultLogger.With("error", err).Error("Error creating operator")
-        panic(err)
+    // Set up tracing
+    if cfg.OTelConfig.Host != "" {
+        simple.SetTraceProvider(simple.OpenTelemetryConfig{
+            Host:        cfg.OTelConfig.Host,
+            Port:        cfg.OTelConfig.Port,
+            ConnType:    simple.OTelConnType(cfg.OTelConfig.ConnType),
+            ServiceName: cfg.OTelConfig.ServiceName,
+        })
     }
 
-    // Wrap our resource watchers in OpinionatedWatchers, then add them to the controllera
-    {{ $p := .WatcherPackage }}{{ range $key, $val := .GVToKind }}{{ range $val }}{{.MachineName}}Watcher, err := {{if ne $p ""}}{{$p}}.{{end}}New{{.Kind}}Watcher()
-    if err != nil {
-        logging.DefaultLogger.With("error", err).Error("Unable to create {{.Kind}}Watcher")
-        panic(err)
+    // Create the operator config and the runner
+    operatorConfig := operator.RunnerConfig{
+        KubeConfig: kubeConfig.RestConfig,
+        WebhookConfig: operator.RunnerWebhookConfig{
+            Port: cfg.WebhookServer.Port,
+            TLSConfig: k8s.TLSConfig{
+                CertPath: cfg.WebhookServer.TLSCertPath,
+                KeyPath:  cfg.WebhookServer.TLSKeyPath,
+            },
+        },
+        MetricsConfig: operator.RunnerMetricsConfig{
+            Enabled: true,
+        },
     }
-    err = runner.WatchKind({{if $.KindsAreGrouped }}{{$.ToPackageNameVariable ($key.String)}}.{{.Kind}}Kind(){{else}}{{ $.ToPackageName .MachineName }}.Kind(){{end}}, {{.MachineName}}Watcher, operator.ListWatchOptions{
-        Namespace: resource.NamespaceAll,
-    })
+    runner, err := operator.NewRunner(operatorConfig)
     if err != nil {
-        logging.DefaultLogger.With("error", err).Error("Error adding {{.Kind}} watcher to controller")
+        logging.DefaultLogger.With("error", err).Error("Unable to create operator runner")
         panic(err)
     }
-    {{ end }}{{ end }}
 
-    stopCh := make(chan struct{})
+    // Context and cancel for the operator's Run method
+    ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Very small nit, but since we've moved to contexts from raw channels, how about using:
```go
ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
```
and getting rid of the low-level signal handling and extra goroutine?